File: src/main/kotlin/com/smarthome/SmartHome/repository/UserRepository.java
package com.smarthome.SmartHome.repository;

import com.smarthome.SmartHome.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    User findUserByLogin(String login);
}

File: src/main/kotlin/com/smarthome/SmartHome/repository/UserRepository.kt
package com.smarthome.smarthome.repository

import com.smarthome.smarthome.repository.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long?> {
    fun findUserByLogin(login: String?): User
}
